resources/interactive.mjs

```diff
@@ -0,0 +1,186 @@
+    const checkboxes = [];
+    for (let suiteIndex = 0; suiteIndex < suites.length; suiteIndex++) {
+        const suite = suites[suiteIndex];
+        const li = document.createElement('li');
+        const checkbox = document.createElement('input');
+        checkbox.id = suite.name;
+        checkbox.type = 'checkbox';
+        checkbox.checked = !suite.disabled;
+        checkbox.onchange = () => { suite.disabled = !checkbox.checked; };
+        checkbox.onchange();
+        checkboxes.push(checkbox);
+
+        li.appendChild(checkbox);
+        var label = document.createElement('label');
+        label.appendChild(document.createTextNode(formatTestName(suite.name)));
+        li.appendChild(label);
+        label.htmlFor = checkbox.id;
+
+        const testList = document.createElement('ol');
+        for (let testIndex = 0; testIndex < suite.tests.length; testIndex++) {
+            const testItem = document.createElement('li');
+            const test = suite.tests[testIndex];
+            const anchor = document.createElement('a');
+            anchor.id = suite.name + '-' + test.name;
+            test.anchor = anchor;
+            anchor.appendChild(document.createTextNode(formatTestName(suite.name, test.name)));
+            testItem.appendChild(anchor);
+            testList.appendChild(testItem);
+        }
+        li.appendChild(testList);
+
+        ol.appendChild(li);
+    }
+
+    control.appendChild(ol);
+
+    let button = document.createElement('button');
+    button.textContent = 'Step';
+    button.onclick = onStep;
+    control.appendChild(button);
+
+    button = document.createElement('button');
+    button.textContent = 'Run';
+    button.id = 'runSuites';
+    button.onclick = onRunSuites;
+    control.appendChild(button);
+
+    button = document.createElement('button');
+    button.textContent = 'Select all';
+    button.onclick = () => {
+        for (var suiteIndex = 0; suiteIndex < suites.length; suiteIndex++) {
+            suites[suiteIndex].disabled = false;
+            checkboxes[suiteIndex].checked = true;
+        }
+    };
+    control.appendChild(button);
+
+    button = document.createElement('button');
+    button.textContent = 'Unselect all';
+    button.onclick = () => {
+        for (var suiteIndex = 0; suiteIndex < suites.length; suiteIndex++) {
+            suites[suiteIndex].disabled = true;
+            checkboxes[suiteIndex].checked = false;
+        }
+
+    };
+    control.appendChild(button);
+
+    return control;
+}
+
+const parseQueryString = (function (pairList)
+{
+    const pairs = {};
+    for (let i = 0; i < pairList.length; ++i) {
+        const keyValue = pairList[i].split('=', 2);
+        if (keyValue.length == 1)
+            pairs[keyValue[0]] = '';
+        else
+            pairs[keyValue[0]] = decodeURIComponent(keyValue[1].replace(/\+/g, ' '));
+    }
+    return pairs;
+})(window.location.search.substr(1).split('&'));
+
+function disableAllSuitesExcept(suiteName)
+{
```

Comment:
Nit: I think this styling is inconsistent? I don't particularly care or even have the firmest grasp on what the styling rules are, but once we actually get the code formatting stuff landed it would be nice to minimize noise in the blame record from the autoformatting step.

Reply:
From https://webkit.org/code-style-guidelines/#braces i think functions should have braces on new lines (I might be wrong)

Reply:
Yes. Functions should have braces on new line.